stl/inc/type_traits

```diff
@@ -2360,5 +2366,10 @@ _STD_END
 _STL_RESTORE_CLANG_WARNINGS
 #pragma warning(pop)
 #pragma pack(pop)
+
+#ifdef __clang__
+#pragma clang diagnostic pop

```

Comment:
The MSVC pop which corresponds to this Clang pop is on line 1679. Could we move this one closer?